libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -724,10 +724,30 @@ void CodeTransform::visitExpression(Expression const& _expression)
 
 void CodeTransform::visitStatements(vector<Statement> const& _statements)
 {
+	auto fdSeries = groupFunctionDefinitions(_statements);
+
+	const AbstractAssembly::LabelID* seriesLabel = nullptr;
+	const Statement* seriesEnd = nullptr;
 	for (auto const& statement: _statements)
 	{
+		if (fdSeries[&statement])
+		{
+			auto label = m_assembly.newLabelId();
+			m_assembly.appendJumpTo(label, 0);
+
+			seriesLabel = &label;
```

Comment:
I think the error is here, you are storing the address of a stack variable and then it goes out of scope.